src/viam/components/resource_manager.py

```diff
@@ -39,6 +42,12 @@ def register(self, component: ComponentBase):
             if hasattr(subtype, "get_resource_name"):
                 rn = subtype.get_resource_name(component.name)  # type: ignore
                 rnames[rn] = component
+            if ":" in rn.name:
```

Comment:
It's entirely possible that `rn` here doesn't exists (since it's created above within an `if` statement). 

It's probably better to move this to a separate `for` loop once the `rnames` dict is done populating.